workflow/controller/controller.go

```diff
@@ -225,6 +225,7 @@ func NewWorkflowController(ctx context.Context, restConfig *rest.Config, kubecli
 			PodPhase:          wfc.getPodPhaseMetrics,
 			WorkflowPhase:     wfc.getWorkflowPhaseMetrics,
 			WorkflowCondition: wfc.getWorkflowConditionMetrics,
+			LeaderState:       wfc.IsLeader,
```

Comment:
why not name `LeaderState` as `IsLeader` for consistency? Because of the `0`/`1` gauge? the type should make that inferable. Also already have `addIsLeader` as well

Reply:
Yeah, that's a decent call